src/arrayMethodSort.js

```diff
@@ -4,9 +4,39 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(compareFunction = compareAsStrings) {
+    let count;
+
+    do {
+      count = 0;
+
+      for (let i = 1; i < this.length; i++) {
+        const prev = this[i - 1];
+        const current = this[i];
+
+        if (compareFunction(prev, current) > 0) {
+          this[i - 1] = current;
+          this[i] = prev;
+          count++;
+        }
+      }
+    } while (count > 0);
+
+    return this;
   };
 }
 
+const compareAsStrings = (a, b) => {
+  const stringA = String(a);
+  const stringB = String(b);
+
+  if (stringA > stringB) {
+    return 1;
+  } else if (stringA === stringB) {
+    return 0;
+  } else {
+    return -1;
+  }
```

Comment:
```suggestion
  return stringA.localeCompare(stringB);
```

Reply:
it won't work in my case because the logic is build on the compareFunction should return a value greater than zero, but not true or false value

Reply:
@xiimiik Also default sort method compares strings by charcode, not like localeCompare
<img width="369" alt="Screenshot 2022-09-20 at 17 57 29" src="https://user-images.githubusercontent.com/84275496/191516523-7f35b4c2-ccc8-4cad-89d4-2cd856567bb4.png">